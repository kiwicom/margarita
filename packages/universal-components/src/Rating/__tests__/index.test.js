// @flow

import * as React from 'react';
import { render } from 'react-native-testing-library';
import snapshotDiff from 'snapshot-diff';

import { Rating } from '..';

describe('Rating', () => {
  const rating = 3;
  const style = { color: '#ffb100' };

  const { getByText, getAllByProps } = render(
    <Rating style={style} rating={rating} />,
  );

  it('should contain stars', () => {
    expect(getByText('â˜…'.repeat(rating))).toBeDefined();
  });

  it('should have passed props', () => {
    expect(
      getAllByProps({
        rating,
        style,
      }),
    ).toBeDefined();
  });

  it('should match snapshot diff', () => {
    const base = <Rating />;
    const extend = <Rating style={style} rating={rating} />;
    expect(snapshotDiff(base, extend)).toMatchSnapshot();
  });
});
