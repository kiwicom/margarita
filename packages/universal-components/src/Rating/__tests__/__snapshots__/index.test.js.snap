// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating should match snapshot diff 1`] = `
"Snapshot Diff:
- <Rating />
+ <Rating rating={3} style={{\\"color\\": \\"#ffb100\\"}} />

@@ -7,12 +7,14 @@
        },
        Array [
          Object {
            \\"letterSpacing\\": 2,
          },
-         undefined,
+         Object {
+           \\"color\\": \\"#ffb100\\",
+         },
        ],
      ]
    }
  >
-   
+   ★★★
  </Text>"
`;
